Python/chapter03/p02_stack_min/miguelHx.py

```diff
@@ -0,0 +1,222 @@
+"""
+Python version 3.7.0
+3.2 - Stack Min
+How would you design a stack which, in addition to
+push and pop, has a function min which returns the minimum element?
+Push, pop and min should all operate in O(1) time.
+"""
+
+import copy
+import unittest
+import sys
+from typing import List
+
+
+class StackNode(object):
+
+    def __init__(self, data):
+        self.data = data
+        self.next = None # next is a pointer to a StackNode object
+        self.running_min = sys.maxsize
+
+class MyStack(object):
+    """
+    Stack data structure implementation.
+    Uses LIFO (last-in first-out) ordering.
+    The most recent item added to the stack is
+    the first removed.  Traversal is top to bottom.
+    """
+
+    def __init__(self):
+        self.top = None # top is a pointer to StackNode object
```

Comment:
same suggestions w.r.t. typing as with https://github.com/techqueria/data-structures-and-algorithms/pull/82

Reply:
Updated here 64d2effe52c84e024dc164a6e2b2e9356d485e21